kore/src/Kore/JsonRpc.hs

```diff
@@ -180,10 +181,13 @@ respond serverState moduleName runSMT =
                                     ++ [ Rewrite
                                         { result =
                                             Success
-                                                { rewrittenTerm = Nothing
+                                                { rewrittenTerm =
+                                                    PatternJson.fromTermLike . Pattern.term
+                                                        <$> successfullyRewrittenTerm
```

Comment:
That seems wrong... we would emit all the rewrite traces with the same rewritten term (the final result).
The rewritten/original terms must be stored in the `Exec.RuleTrace`  if we want to emit them (i.e., the change needs to go deeper into the rewriting logic).

Reply:
Right. Good catch. I though I could get away with a simple solution, but did not give it enough thought. Thanks!